src/vban-output.c

```diff
@@ -105,6 +107,61 @@ static bool update_string(char **opt, obs_data_t *settings, const char *name)
 	return false;
 }
 
+static void resolve_finalize(struct vban_out_s *v, const struct in_addr *addr)
+{
+	if (!v->rt)
+		return;
+
+	if (addr)
+		v->ip_to = *addr;
+
+	resolve_thread_release(v->rt);
+	v->rt = NULL;
+}
+
+static void resolve_succeeded(void *data, const struct in_addr *addr)
+{
+	struct vban_out_s *v = data;
+	blog(LOG_DEBUG, "Resolved host address: %s", inet_ntoa(*addr));
+	resolve_finalize(v, addr);
+}
+
+static void resolve_failed(void *data)
+{
+	struct vban_out_s *v = data;
+	blog(LOG_DEBUG, "Failed to resolve");
+	resolve_finalize(v, NULL);
+}
+
+static void vban_out_update_ip_to(struct vban_out_s *v, const char *ip_to)
+{
+	if (v->rt) {
+		resolve_thread_set_callbacks(v->rt, NULL, NULL, NULL);
+		resolve_thread_release(v->rt);
+		v->rt = NULL;
+	}
+
+	if (!ip_to || !*ip_to) {
+		v->ip_to.s_addr = 0xFFFFFFFF;
+		return;
+	}
+
+	struct in_addr addr = {0};
+	if (inet_pton(AF_INET, ip_to, &addr)) {
+		v->ip_to.s_addr = addr.s_addr;
+		return;
+	}
+
+	v->rt = resolve_thread_create(ip_to);
+	if (!v->rt)
+		return;
+
+	blog(LOG_DEBUG, "Resolving host name '%s'", ip_to);
+	resolve_thread_set_callback_mutex(v->rt, &v->mutex);
```

Comment:
This will cause a dangling pointer access if the resolving thread takes longer than the lifetime of `vban_out_s`.